process/interceptors/multiDataInterceptor.go

```diff
@@ -152,12 +157,19 @@ func (mdi *MultiDataInterceptor) ProcessReceivedMessage(message p2p.MessageP2P,
 	for index, dataBuff := range multiDataBuff {
 		var interceptedData process.InterceptedData
 		interceptedData, err = mdi.interceptedData(dataBuff, message.Peer(), fromConnectedPeer)
+
 		listInterceptedData[index] = interceptedData
 		if err != nil {
 			mdi.throttler.EndProcessing()
 			return err
 		}
 
+		errCache := mdi.checkIfMessageHasBeenProcessed(interceptedData)
```

Comment:
We could solve this if we save also the verification status inside the cache.
In this case we might still need the mutex per message key, so that we do not process it twice while not having the status inside the cache.